test/Microsoft.AspNetCore.Mvc.Razor.Test/TagHelpers/TagHelperComponentPropertyActivatorTest.cs

```diff
@@ -0,0 +1,73 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.IO;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc.Abstractions;
+using Microsoft.AspNetCore.Mvc.ModelBinding;
+using Microsoft.AspNetCore.Mvc.Razor.Internal;
+using Microsoft.AspNetCore.Mvc.Rendering;
+using Microsoft.AspNetCore.Mvc.ViewEngines;
+using Microsoft.AspNetCore.Mvc.ViewFeatures;
+using Microsoft.AspNetCore.Razor.TagHelpers;
+using Microsoft.AspNetCore.Routing;
+using Moq;
+using Xunit;
+
+namespace Microsoft.AspNetCore.Mvc.Razor.Test.TagHelpers
+{
+    public class TagHelperComponentPropertyActivatorTest
+    {
+        [Fact]
+        public void InitializesViewContext()
+        {
+            // Arrange
+            var tagHelperComponent = new TestTagHelperComponent();
+            var viewContext = CreateViewContext(new DefaultHttpContext());
+            var viewDataValue = "Value";
```

Comment:
No need to modify viewdata. Since you're verifying that it's the `Same` type all of the underlying properties are guarunteed to be the same as well.